### ISSUE TITLE
CVE-2022-24765 : Unexpected denial for user part of well-known SID 'Administrators'

### ISSUE THREAD

**Issue by nomennescio** [git-for-windows/git#3955]
- [x] I was not able to find an [open](https://github.com/git-for-windows/git/issues?q=is%3Aopen) or [closed](https://github.com/git-for-windows/git/issues?q=is%3Aclosed) issue matching what I'm seeing

### Setup

 - Which version of Git for Windows are you using? Is it 32-bit or 64-bit?

```
git version 2.37.1.windows.1
cpu: x86_64
built from commit: 323a69709944b193bb5cee81ff09fe9a4a686df5
sizeof-long: 4
sizeof-size_t: 8
shell-path: /bin/sh
feature: fsmonitor--daemon
```

 - Which version of Windows are you running? Vista, 7, 8, 10? Is it 32-bit or 64-bit?

```
Microsoft Windows [Version 10.0.18363.1916]
```

 - What options did you set as part of the installation? Or did you choose the
   defaults?

```
$  cat /etc/install-options.txt
Editor Option: CustomEditor
Custom Editor Path: <xxxx>
Default Branch Option:
Path Option: BashOnly
Plink Path: <xxxx>
SSH Option: Plink
Tortoise Option: false
CURL Option: OpenSSL
CRLF Option: CRLFCommitAsIs
Bash Terminal Option: MinTTY
Git Pull Behavior Option: Merge
Use Credential Manager: Disabled
Performance Tweaks FSCache: Enabled
Enable Symlinks: Enabled
Enable Pseudo Console Support: Disabled
Enable FSMonitor: Enabled
```

- Any other interesting things about your environment that might be related
   to the issue you're seeing?

Issue does not happen when running from Cygwin Bash

 - Which terminal/shell are you running Git from? e.g Bash/CMD/PowerShell/other

Git GUI Bash

 - What commands did you run to trigger this issue? If you can provide a
   [Minimal, Complete, and Verifiable example](http://stackoverflow.com/help/mcve)
   this will help us understand the issue.

```
git status
```

 - What did you expect to occur after running these commands?

no issue

 - What actually happened instead?

```
 git status
warning: 'C:/xxxx' is owned by:
        'S-1-5-32-544'
but the current user is:
        'S-1-5-21-dddd-xxxx-yyy-zzzz'
fatal: detected dubious ownership in repository at 'C:/dev/programs/language/factor'
```

'S-1-5-32-544' is the local Administrators well-known SID.
'S-1-5-21-dddd-xxxx-yyy-zzzz' is part of that group, hence should have no issue accessing that directory

**Comment by rimrul** [comment]
near duplicate of #3798

**Comment by nomennescio** [comment]
I've read issue #3798, which indeed has overlap. This issue however is more strict; SID Administrators is a special group, anyone part of it can do anything to the system. However on Windows it is BAD practice to set ownership of a directory or file to an individual user in case of NTFS files that need modifiable access by all members of that group. One should instead use 'SID Administrator' as owner, which will then be inherited on child objects (at least in the regular case). Therefore, this is a perfectly legal, common, and safe use case, which should not be flagged as an error (for security reasons nonetheless!).

**Comment by nomennescio** [comment]
Furthermore, it's interesting that executing the same (!) git executable under Cygwin Bash does not report any error at all.

**Comment by nomennescio** [comment]
> > Furthermore, it's interesting that executing the same (!) git executable under Cygwin Bash does not report any error at all.
> 
> That's very unlikely. This sounds like it's not actually the same git executable.

Running from within Cygwin Bash:
```
$ git --version --build-options
git version 2.37.1.windows.1
cpu: x86_64
built from commit: 323a69709944b193bb5cee81ff09fe9a4a686df5
sizeof-long: 4
sizeof-size_t: 8
shell-path: /bin/sh
feature: fsmonitor--daemon
```

and `which` git points to the same binary

**Comment by nomennescio** [comment]
It most likely has to do with how Cygwin maps SIDs.

**Comment by rimrul** [comment]
> (Note that this is actually a *likely* scenario, as Git for Windows installs itself into `PATH`, which is by default used by Cygwin)

No. What's unlikely isn't that you're able to call the same executable from cygwin, but that it would behave any differently when called from cygwin.

**Comment by rimrul** [comment]
> It most likely has to do with how Cygwin maps SIDs. See e.g. https://cygwin.com/cygwin-ug-net/ntsec.html

That's only relevant to executables linked against cygwin. Not for our `git.exe`.

**Comment by rimrul** [comment]
> SID Administrators is a special group

Note that there is a proposal on the `git` mailing list to give special treatment to the administrators group:

https://lore.kernel.org/git/be06d711a1349a886cad63f8f2ec6675e6d5055e.1657700238.git.gitgitgadget@gmail.com/

**Comment by nomennescio** [comment]
> > (Note that this is actually a _likely_ scenario, as Git for Windows installs itself into `PATH`, which is by default used by Cygwin)
> 
> No. What's unlikely isn't that you're able to call the same executable from cygwin, but that it would behave any differently when called from cygwin.

Still it does

**Comment by rimrul** [comment]
> > > It most likely has to do with how Cygwin maps SIDs. See e.g. https://cygwin.com/cygwin-ug-net/ntsec.html
> > 
> > That's only relevant to executables linked against cygwin. Not for our `git.exe`.
> 
> It is, because I'm running the same executable under Cygwin

You running it under cygwin does not mean it's linked against cygwin.

**Comment by nomennescio** [comment]
> > > > It most likely has to do with how Cygwin maps SIDs. See e.g. https://cygwin.com/cygwin-ug-net/ntsec.html
> > > 
> > > 
> > > That's only relevant to executables linked against cygwin. Not for our `git.exe`.
> > 
> > 
> > It is, because I'm running the same executable under Cygwin
> 
> You running it under cygwin does not mean it's linked against cygwin.

It isn't. I don't claim that. I don't see how it's relevant.
I run the same executable, it fails under Git Bash, and succeeds under Cygwin Bash.

**Comment by rimrul** [comment]
If it isn't linked against cygwin it isn't affected by cygwins SID mapping.

**Comment by dscho** [comment]
This should be fixed by https://github.com/git/git/commit/3f7207e2ea9.

**Comment by Juma-creator** [comment]
git config --global --add safe.directory C:/path/to/repo
